cpp/dolfinx/fem/FiniteElement.cpp

```diff
@@ -164,25 +163,18 @@ void FiniteElement::evaluate_reference_basis(
   assert(_element);
   xt::xtensor<double, 4> basis = _element->tabulate(0, X);
   assert(basis.shape(1) == X.shape(0));
-  for (std::size_t p = 0; p < basis.shape(1); ++p)
-  {
-    for (std::size_t d = 0; d < basis.shape(2); ++d)
-    {
-      for (std::size_t v = 0; v < basis.shape(3); ++v)
-        reference_values(p, d, v) = basis(0, p, d, v);
-    }
-  }
+  reference_values = xt::view(basis, 0, xt::all(), xt::all(), xt::all());
 }
 //-----------------------------------------------------------------------------
-// void FiniteElement::evaluate_reference_basis_derivatives(
-//     std::vector<double>& /*values*/, int /*order*/,
-//     const xt::xtensor<double, 2>& /*X*/) const
-// {
-//   // NOTE: This function is untested. Add tests and re-active
-//   throw std::runtime_error(
-//       "FiniteElement::evaluate_reference_basis_derivatives required
-//       updating");
-// }
+void FiniteElement::evaluate_reference_basis_derivatives(
```

Comment:
The function allocates and copies the data to the input.
Wouldn't it be simpler to return it?